--- a/Makefile.in	2017-10-09 15:47:26.577014210 +0800
+++ b/Makefile.in	2017-10-09 16:02:39.415045650 +0800
@@ -10,7 +10,6 @@ CPP = @CPP@
 
 INSTALL = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
-INSTALL_DATA = @INSTALL_DATA@
 
 DEFS = @DEFS@
 LIBS = @LIBS@
@@ -20,19 +19,16 @@ CFLAGS = @CFLAGS@
 LDSTATIC = 
 LDFLAGS = @LDFLAGS@
 
+DESTDIR = @DESTDIR@
+
 SHELL_PROG = ksh
 
 prefix = @prefix@
-exec_prefix = @exec_prefix@
-transform=@program_transform_name@
 
-bindir = $(exec_prefix)/bin
-mandir = $(prefix)/man/man$(manext)
+bindir = /bin
+mandir = $(prefix)/share/man/man$(manext)
 manext = 1
 
-# Suffix for executables: nothing for unix, .exe for os/2.
-exe_suffix=
-
 SHELL = /bin/sh
 
 SRCS = alloc.c c_ksh.c c_sh.c c_test.c c_ulimit.c edit.c emacs.c \
@@ -70,7 +66,7 @@ OS2FILES = os2/Makefile os2/config.h os2
 	os2/kshrc.ksh os2/make.sed os2/os2.c os2/os2siglist.out \
 	os2/README.os2 os2/NEWS.os2 os2/os2bugs os2/th.cmd os2/config.cache
 
-all: $(SHELL_PROG)$(exe_suffix) 
+all: $(SHELL_PROG)
 
 # This shouldn't be first - some makes don't know about PRECIOUS and assume it
 # is the default target.
@@ -79,27 +75,23 @@ all: $(SHELL_PROG)$(exe_suffix)
 .c.o:
 	$(CC) -c $(CPPFLAGS) $(DEFS) -I. -I$(srcdir) $(CFLAGS) $<
 
-install: installdirs all
-	$(INSTALL_PROGRAM) $(SHELL_PROG)$(exe_suffix) $(bindir)/`echo $(SHELL_PROG)|sed '$(transform)'`$(exe_suffix)
-	-$(INSTALL_DATA) $(SHELL_PROG).1 $(mandir)/`echo $(SHELL_PROG)|sed '$(transform)'`.$(manext)
-	-@prog=$(bindir)/`echo $(SHELL_PROG)|sed '$(transform)'`$(exe_suffix);\
+install:
+	$(INSTALL_PROGRAM) -D $(SHELL_PROG) $(DESTDIR)$(bindir)/$(SHELL_PROG)
+	$(INSTALL_PROGRAM) -D -m 0644 $(SHELL_PROG).$(manext) $(DESTDIR)$(mandir)/$(SHELL_PROG).$(manext)
 	    test -f /etc/shells \
 	    && (grep "^$$prog\$$" /etc/shells > /dev/null \
 		|| echo \
 "NOTE: /etc/shells does not contain $$prog \
       you should add it if you want to set your shell to $(SHELL_PROG)")
 
-installdirs:
-	$(srcdir)/mkinstalldirs $(bindir) $(mandir)
-
 uninstall:
-	rm -f $(bindir)/`echo $(SHELL_PROG)|sed '$(transform)'`$(exe_suffix)
-	rm -f $(mandir)/`echo $(SHELL_PROG)|sed '$(transform)'`.$(manext)
+	rm -f $(DESTDIR)$(bindir)/$(SHELL_PROG)
+	rm -f $(DESTDIR)$(mandir)/$(SHELL_PROG).$(manext)
 
 check test:
-	$(srcdir)/tests/th-sh $(srcdir)/tests/th -s $(srcdir)/tests -p ./$(SHELL_PROG)$(exe_suffix) -C pdksh,sh,ksh,posix,posix-upu $(TESTARGS)
+	$(srcdir)/tests/th-sh $(srcdir)/tests/th -s $(srcdir)/tests -p ./$(SHELL_PROG) -C pdksh,sh,ksh,posix,posix-upu $(TESTARGS)
 
-$(SHELL_PROG)$(exe_suffix): $(OBJS)
+$(SHELL_PROG): $(OBJS)
 	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LIBS)
 
 
@@ -132,17 +124,17 @@ siglist.out: config.h sh.h siglist.in si
 	$(srcdir)/siglist.sh "$(CPP) $(CPPFLAGS) $(DEFS) -I. -I$(srcdir)" < $(srcdir)/siglist.in > tmpsiglist.out
 	mv tmpsiglist.out siglist.out
 
-debugtools: check-fd$(exe_suffix) check-sigs$(exe_suffix) \
-	check-pgrp$(exe_suffix)
+debugtools: check-fd check-sigs \
+	check-pgrp
 
 check-fd.o check-pgrp.o check-sigs.o: config.h
-check-fd$(exe_suffix): check-fd.o
+check-fd: check-fd.o
 	$(CC) $(LDFLAGS) -o $@ check-fd.o $(LIBS)
 
-check-pgrp$(exe_suffix): check-pgrp.o
+check-pgrp: check-pgrp.o
 	$(CC) $(LDFLAGS) -o $@ check-pgrp.o $(LIBS)
 
-check-sigs$(exe_suffix): check-sigs.o
+check-sigs: check-sigs.o
 	$(CC) $(LDFLAGS) -o $@ check-sigs.o $(LIBS)
 
 TAGS: $(SRCS) $(HDRS)
@@ -152,10 +144,10 @@ tags: $(SRCS) $(HDRS)
 	cd $(srcdir) && ctags -wt $(SRCS) $(HDRS)
 
 clean:
-	rm -f ksh$(exe_suffix) sh$(exe_suffix) $(OBJS) siglist.out \
+	rm -f ksh sh $(OBJS) siglist.out \
 	      core a.out mon.out gmon.out \
-	      version.c.bak Makefile.bak Makefile.tmp check-fd$(exe_suffix) \
-	      check-pgrp$(exe_suffix) check-sigs$(exe_suffix)
+	      version.c.bak Makefile.bak Makefile.tmp check-fd \
+	      check-pgrp check-sigs
 
 mostlyclean: clean
 
